Frontend_WhatToWeEat/src/router/index.js

```diff
@@ -0,0 +1,17 @@
+import { createRouter, createWebHashHistory } from 'vue-router'
+import Login from '../pages/Login.vue'
+import Register from '../pages/Register.vue'
+
+const routes = [
+  {path:'/', component: Login},
+  { path: '/login', component: Login },
+  { path: '/register', component: Register },
+]
+
+const router = createRouter({
+  // 4. Provide the history implementation to use. We are using the hash history for simplicity here.
+  history: createWebHashHistory(),
```

Comment:
我在思考 `WebHashHistory` 和 `WebHistory` 哪個會比較好
用 hash history 有什麼考量嗎

Reply:
https://router.vuejs.org/guide/essentials/history-mode.html#html5-mode

裡面是這麼說的

> When using createWebHistory(), the URL will look "normal," e.g. https://example.com/user/id. Beautiful!
Here comes a problem, though: Since our app is a single page client side app, without a proper server configuration, the users will get a 404 error if they access https://example.com/user/id directly in their browser. Now that's ugly.
Not to worry: To fix the issue, all you need to do is add a simple catch-all fallback route to your server. 
If the URL doesn't match any static assets, it should serve the same index.html page that your app lives in. Beautiful, again!

用Webhistory似乎要對server做甚麼配置才不會出錯，這我還不是很清楚怎麼弄
如果說只是暫時能run就好，就想說先用Hash，但是網址都會有一個#符號